activesupport/test/hash_with_indifferent_access_test.rb

```diff
@@ -536,6 +536,18 @@ def test_nested_dig_indifferent_access
     assert_equal 1234, data.dig(:this, :views)
   end
 
+  def test_argless_default_with_existing_nil_key
+    h = Hash.new(:default)
+    h[nil] = "defined"
+    h = h.with_indifferent_access
+    assert_equal :default, h.default
+  end
+
+  def test_default_with_argument
+    h = Hash.new { 5 }.merge(1 => 2).with_indifferent_access
+    assert_equal 5, h.default(1)
+  end
```

Comment:
Maybe we need a test for the original issue. In the original issue there is no mention to default. Only the nil key was set in the hash and later tried to read a random key and got the value equivalent to `nil`.